app/components/autocomplete/slash_suggestion/slash_suggestion.js

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {
+    Alert,
+    FlatList,
+    Text,
+    TouchableOpacity
+} from 'react-native';
+import {intlShape} from 'react-intl';
+
+import {RequestStatus} from 'mattermost-redux/constants';
+
+import {makeStyleSheetFromTheme, changeOpacity} from 'app/utils/theme';
+
+const SLASH_REGEX = /(^\/)([a-zA-Z-]*)$/;
+const ROW_HEIGHT = 55;
+
+export default class SlashSuggestion extends Component {
+    static propTypes = {
+        actions: PropTypes.shape({
+            getAutocompleteCommands: PropTypes.func.isRequired
+        }).isRequired,
+        currentTeamId: PropTypes.string.isRequired,
+        commands: PropTypes.array,
+        commandsRequest: PropTypes.object.isRequired,
+        executeCommandRequest: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        onChangeText: PropTypes.func.isRequired,
+        onResultCountChange: PropTypes.func.isRequired,
+        value: PropTypes.string
+    };
+
+    static defaultProps = {
+        defaultChannel: {},
+        value: ''
+    };
+
+    static contextTypes = {
+        intl: intlShape
+    }
+
+    state = {
+        active: false,
+        suggestionComplete: false,
+        dataSource: [],
+        lastCommandRequest: 0
+    };
+
+    componentWillReceiveProps(nextProps) {
+        const {currentTeamId, executeCommandRequest} = this.props;
+        const {
+            commands: nextCommands,
+            commandsRequest: nextCommandsRequest,
+            currentTeamId: nextTeamId,
+            executeCommandRequest: nextExecuteCommandRequest,
+            value: nextValue
+        } = nextProps;
+
+        if (nextExecuteCommandRequest.status === RequestStatus.FAILURE && executeCommandRequest.status === RequestStatus.STARTED) {
```

Comment:
I would say that this is not needed here and you could easily move this to the action where the command is being executed, await for it and if it returns and error, you can display the alert and even use the error that was returned